test/e2e/serial/tests/configuration.go

```diff
@@ -736,6 +739,41 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
 			nrsGot := objRefListToStringList(nroSchedObj.Status.RelatedObjects)
 			Expect(nrsGot).To(Equal(nrsExpected), "mismatching related objects for NUMAResourcesScheduler")
 		})
+
+		It("[slow][tier1] ignores non-matching kubeletconfigs", func(ctx context.Context) {
+			By("getting the NROP object")
+			nroOperObj := &nropv1.NUMAResourcesOperator{}
+			nroKey := objects.NROObjectKey()
+			err := fxt.Client.Get(ctx, nroKey, nroOperObj)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", nroKey.String())
+
+			By("recording the current kubeletconfig and configmap status")
+			kcCmsPre, err := getKubeletConfigMapsSoftOwnedBy(ctx, fxt.Client, nroOperObj.Name)
+			Expect(err).ToNot(HaveOccurred(), "cannot list KubeletConfig ConfigMaps in the cluster (PRE)")
+			kcCmNamesPre := sets.List[string](accumulateKubeletConfigNames(kcCmsPre))
+			klog.Infof("initial set of configmaps from kubeletconfigs: %v", strings.Join(kcCmNamesPre, ","))
+
+			By("creating extra ctrplane kubeletconfig")
+			ctrlPlaneKc := intobjs.NewKubeletConfigAutoresizeControlPlane()
+			err = fxt.Client.Create(ctx, ctrlPlaneKc)
```

Comment:
does this trigger an update on the controlplane nodes? if so I believe we better wait for that update to finish here and in defer